default/ship_parts.txt

```diff
@@ -1134,11 +1134,22 @@ Count condition = And [
             ]
 '''
 
+// Beginning of Stacking macros
+/* For the AI and general uniformity, shields, detectors and cloaks are assigned a capacity.
+However, to prevent stacking, for each group X there is a X_STACKING macro which basically 
+first removes all effects of the used parts (by subtracting the summed capacity of all used parts)
```

Comment:
this ridiculous workaround is why I think that capacity-based effects should not be automatically added to parts, or at least there should be a way to disable them in the part script.